code/modules/surgery/blood_filter.dm

```diff
@@ -60,3 +80,21 @@
 		"<span class='warning'>[user] screws up, brusing [target]'s chest!</span>",
 		"<span class='warning'>[user] screws up!</span>")
 	target.adjustBruteLoss(5)
+
+/datum/surgery/blood_filter/upgraded
+	name = "Filter Blood (Adv.)"
+	filtering_step_type = /datum/surgery_step/filter_blood/upgraded
+	replaced_by = /datum/surgery/blood_filter/femto
+
+/datum/surgery_step/filter_blood/upgraded
+	time = 1.85 SECONDS
+	tox_heal_factor = 0.075
+
+/datum/surgery/blood_filter/femto
+	name = "Filter Blood (Exp.)"
+	filtering_step_type = /datum/surgery_step/filter_blood/femto
+	replaced_by = null
+
+/datum/surgery_step/filter_blood/femto
+	time = 1 SECONDS
+	tox_heal_factor = 0.15
```

Comment:
if you're decreasing speed, it doesn't have to be higher factor, because increased speed itself provides higher healing efficiency.